### PR TITLE
New Auth Flow: Mk2

### PR DESCRIPTION
**Pull request by jleandroperez**
### Fix
In this PR we're:

- Implementing **SignupRemote** (borrowing @eshurakov 's code!!)
- Implementing **MagicLinkAuthenticator** (same, Evgeny's code)
- Implementing **Signup Requested UI**
- Wiring all the things together!
- Updating the CI secrets

cc @eshurakov 

---

### Test: Preparation!
1. Update your `mobile-secrets` checkout
2. Delete all of the **Simplenote Mac** apps (Electron / AppStore): they'll interfere with the Magic Link handling
3. Make sure you're logged off
4. Install this branch!

### Test: Running / Background
1. Click over the **Sign Up** button
2. Enter your email
3. Click over the **Sign Up** Button

- [x] Verify that shortly after, the SignUp Requested UI shows up
- [x] Verify that clicking **Back** presents the Authentication UI again
- [x] Verify that if you click on the **SignUp Email Link** (**and complete Signup**) the Simplenote Native App ends up logged in

### Test: Running / Background
1. Click over the **Sign Up** button
2. Enter your email
3. Click over the **Sign Up** Button
4. **Close Simplenote**

- [ ] Verify that if you click on the **SignUp Email Link** (**and complete the Signup flow**) the Simplenote Native App ends up logged in

### Test: Regressions
1. Click over the **Log In with WordPress.com** button

- [x] Verify the WordPress SSO flow still works!

### Release
These changes do not require release notes.

### CODE REVIEW

File: Simplenote/AuthViewController+Swift.swift [hunk 1]
```diff
@@ -85,12 +102,67 @@ extension AuthViewController {
...
     @objc
     func clearAuthenticationError() {
         errorField.stringValue = ""
     }
+
+    /// Marks the Username Field as the First Responder
+    ///
+    @objc
+    func ensureUsernameIsFirstResponder() {
+        usernameField?.textField.becomeFirstResponder()
```
Comment (eshurakov) [inline Simplenote/AuthViewController+Swift.swift:124]: Should this be done via `window.makeFirstResponder()`? 🤔

File: Simplenote/SignupVerificationViewController.swift [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        view.window?.transition(to: authViewController)
+    }
+}
+
+
+// MARK: - Localization
+//
+private enum Localization {
+    static let messageTemplate = NSLocalizedString("We’ve sent an email to %1$@. Please check your inbox and follow the instructions.", comment: "Signup Body Text")
+    static let support = NSLocalizedString("Didn’t get an email? Please contact %1$@.", comment: "Signup Support Text")
```
Comment (eshurakov) [inline Simplenote/SignupVerificationViewController.swift]: I believe we wanted to use `Didn't get an email? You may already have an account. Contact %1$@ for help.`

**Comment by eshurakov** [comment]
Something else that I noticed:
* can't click support email in the signup confirmation
* when going back, should we go back to sign up? Now goes to login.

File: Simplenote/AuthViewController+Swift.swift (hunk 1 above)
Comment (jleandroperez) [reply Simplenote/AuthViewController+Swift.swift:124]: Actually this approach (becomeFirstResponder) works well in Big Sur, **BUG** `makeFirstResponder` does not.

Can you pleeease confirm if the Username is set as First Responder, when you switch To / From Signup / LogIn?

Thank yoouuu!

File: Simplenote/SignupVerificationViewController.swift (hunk 2 above)
Comment (jleandroperez) [reply Simplenote/SignupVerificationViewController.swift]: Thanks for the heads up, fixed!!

File: Simplenote/AuthViewController+Swift.swift (hunk 1 above)
Comment (eshurakov) [reply Simplenote/AuthViewController+Swift.swift:124]: Oh, interesting, I thought you're not supposed to directly call `becomeFirstResponder`. But yeah, it works
I noticed that sometimes when switching between login and signup background of active field is not updated.

For example: select password -> switch to sign up -> switch to login:
Screenshot 2021-03-05 at 19 18 59

**Review by eshurakov — APPROVED** [review]
Checked opening the app with the magic link when the app was previously closed. Works good!

**Comment by jleandroperez** [comment]
Thank you for reviewing this @eshurakov !!

Addressing pendings in a follow up PR. For the record, pendings are:

```
1.	can't click support email in the signup confirmation
2.	when going back, should we go back to sign up? Now goes to login.
3.	Oh, interesting, I thought you're not supposed to directly call becomeFirstResponder. 
	But yeah, it works. I noticed that sometimes when switching between login and signup background of active field is not updated.
```


**Thank yooouuu**

**Comment by pachlava** [comment]
[testing] fix/feature verified on MBP 16" Intel | macOS Big Sur 11.2 | Simplenote 2.9 (11223)

**Pull request merged**